From 985b6c709463d61407ada6d6b71643b5d0c835e4 Mon Sep 17 00:00:00 2001
From: Eskil Abrahamsen Blomfeldt <eskil.abrahamsen-blomfeldt@qt.io>
Date: Wed, 17 Sep 2025 09:20:35 +0200
Subject: [PATCH 045/239] Fix Android deployment of VectorImage plugins

When selecting which files to automatically bundle in an Android
package, we look at the dependencies of the app and try to infer
which libraries are needed.

For plugins, which specific ones will be loaded is determined at
runtime, so we include any category of plugin that is used by
one or more of the application's dependencies. This is done by
setting the PLUGIN_TYPES of the module. If the application
depends on this module, it will then include all plugins of said
type. For instance, Qt Gui includes all "imageformats".

For vectorimageformats, we had put this dependency on the
VectorImage import, but since this is loaded at runtime, the
deployment tool was not able to determine the dependency on it.
We instead make it a dependency of Qt Quick instead. VectorImage
is technically a part of Qt Quick and any app depending on it
may want to load vector image formats.

Task-number: QTBUG-140187
Change-Id: I2ab42a6c23710a97936ae39ace47b19411e2c6a8
Reviewed-by: Eirik Aavitsland <eirik.aavitsland@qt.io>
Reviewed-by: Alexey Edelev <alexey.edelev@qt.io>
(cherry picked from commit 43f7c0d5703242eff085a125fc05cdbb39d630ef)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/quick/CMakeLists.txt            | 2 +-
 src/quickvectorimage/CMakeLists.txt | 1 -
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/quick/CMakeLists.txt b/src/quick/CMakeLists.txt
index d3d9087a84..86831e371b 100644
--- a/src/quick/CMakeLists.txt
+++ b/src/quick/CMakeLists.txt
@@ -14,7 +14,7 @@ qt_internal_add_qml_module(Quick
     CLASS_NAME QtQuick2Plugin
     IMPORTS
         QtQml/auto
-    PLUGIN_TYPES scenegraph
+    PLUGIN_TYPES scenegraph vectorimageformats
     SOURCES
         handlers/qquickdragaxis.cpp handlers/qquickdragaxis_p.h
         handlers/qquickdraghandler.cpp handlers/qquickdraghandler_p.h
diff --git a/src/quickvectorimage/CMakeLists.txt b/src/quickvectorimage/CMakeLists.txt
index a179c13603..ab63bc2e39 100644
--- a/src/quickvectorimage/CMakeLists.txt
+++ b/src/quickvectorimage/CMakeLists.txt
@@ -6,7 +6,6 @@
 #####################################################################
 
 qt_internal_add_module(QuickVectorImageGeneratorPrivate
-    PLUGIN_TYPES vectorimageformats
     INTERNAL_MODULE
     SOURCES
         generator/qsvgvisitorimpl_p.h generator/qsvgvisitorimpl.cpp
-- 
2.51.2

